pkg/frontend/querymiddleware/astmapper/sharding.go

```diff
@@ -489,12 +499,13 @@ func (summer *shardSummer) shardAndSquashAggregateExpr(expr *parser.AggregateExp
 		// Create the child expression, which runs the given aggregation operation
 		// on a single shard. We need to preserve the grouping as it was
 		// in the original one.
-		children = append(children, &parser.AggregateExpr{
+		var aggExpr parser.Expr = &parser.AggregateExpr{
```

Comment:
[nit] Can we just do this?

```suggestion
		aggExpr := &parser.AggregateExpr{
```